File: Languages/golang/src/model/UserInfo.go
package userInfo

import (  
    "bytes"
)

type UserInfo struct {  
    _user   string
    _password    string
}

// Metodo privado para encriptar a senha do usuario
// @param password Senha a ser encriptada
// @return Nova senha encriptada
func cryptPassword(password string) string {
    runes := []rune(password)
    for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }

    var buffer bytes.Buffer
    buffer.WriteString("HASH")
    buffer.WriteString(string(runes))
    buffer.WriteString("000")

    return buffer.String()
}

// Atualiza a senha do usuario
// @param password Nova senha do usuario
func (this *UserInfo) SetUser(user string) {
    this._user = user
}

// Atualiza o nome do usuario
// @param user Novo nome do usuario
func (this *UserInfo) setPassword(password string) {
    this._password = password
}

// Obtem o nome do usuario
// @return Nome do usuario
func (this UserInfo) GetUser() string {
    return this._user
}

// Obtem a senha do usuario criptografada
// @return Senha do usuario criptografada
func (this UserInfo) GetPassword() string {
    return cryptPassword(this._password)
}

// Construtor publico da classe
// @param user Nome do usuario
// @param password Senha do usuario    
func New(user string, password string) UserInfo {  
    e := UserInfo {user, password}
    return e
}

File: Languages/dotnet/src/model/UserInfo.cs
namespace experimento.src.model {

    /// <summary>
    /// Informacoes do usuario
    /// </summary>
    public class UserInfo {
        
        /// <summary>
        /// Variável privada para o nome do usuário
        /// </summary>
        private string _user;

        /// <summary>
        /// Variável privada para a senha do usuário
        /// </summary>
        private string _password;
                
        /// <summary>
        /// Variável publica para o nome do usuário
        /// </summary>
        public string user {

            /// <summary>
            /// Obtem o nome do usuario
            /// </summary>
            /// <returns>Nome do usuario</returns>
            get {
                return this._user;
            }

            /// <summary>
            /// Atualiza o nome do usuário
            /// </summary>
            set {
                this._user = value;
            }
        }

        /// <summary>
        /// Variável publica para a senha do usuário
        /// </summary>
        public string password {
            
            /// <summary>
            /// Obtem a senha do usuario criptografada
            /// </summary>
            /// <returns>Senha do usuario criptografada</returns>
            get {
                return cryptPassword(this._password);
            }

            /// <summary>
            /// Atualiza a senha do usuário
            /// </summary>
            set {
                this._password = value;
            }
        }

        /// <summary>
        /// Construtor publico da classe
        /// </summary>
        /// <param name="user">Nome do usuario</param>
        /// <param name="password">Senha do usuario</param>
        public UserInfo(string user, string password) {
            this._user = user;
            this._password = password;
        }

        /// <summary>
        /// Metodo privado para encriptar a senha do usuario
        /// </summary>
        /// <param name="password">Senha a ser encriptada</param>
        /// <returns>Nova senha encriptada</returns>
        private string cryptPassword(string password){
            string[] cryptArray = password.Split("");
            string[] reverseArray = new string[cryptArray.Length];

            for(int index = 0; index < cryptArray.Length; index++){
                reverseArray[index] = cryptArray[cryptArray.Length - index];
            } 

            string crypt = string.Join("",cryptArray);

            return $"HASH{crypt}000";
        }
    }
}